onnxruntime/core/providers/vitisai/imp/graph.cc

```diff
@@ -311,19 +311,30 @@
     }
     auto ORT_MEM_ADDR_tag = process_ext_address(*original_tensor);
     if (!ORT_MEM_ADDR_tag.empty()) {
-      cloned_tensor->set_data_location(ONNX_NAMESPACE::TensorProto_DataLocation_EXTERNAL);
-      auto external_data = cloned_tensor->mutable_external_data();
-      auto p = external_data->Add();
-      *p->mutable_key() = "location";
-      *p->mutable_value() = std::string("<") + graph_ptr;
-    } else if (size >= external_data_threshold) {
-      cloned_tensor->set_data_location(ONNX_NAMESPACE::TensorProto_DataLocation_EXTERNAL);
-      auto external_data = cloned_tensor->mutable_external_data();
-      auto p = external_data->Add();
-      *p->mutable_key() = "location";
-      *p->mutable_value() = std::string("<") + graph_ptr;
+      // ORT 1.23 intializer handling
+      if (size >= external_data_threshold) {
+        cloned_tensor->set_data_location(ONNX_NAMESPACE::TensorProto_DataLocation_EXTERNAL);
+        auto external_data = cloned_tensor->mutable_external_data();
+        auto p = external_data->Add();
+        *p->mutable_key() = "location";
+        *p->mutable_value() = std::string("<") + graph_ptr;
+      } else {
+        // Under threshold: get the data from ORT_MEM_ADDR and add it to cloned graph
```

Comment:
> // Under threshold: get the data from ORT_MEM_AD [](http://example.com/codeflow?start=7&length=49)

Do not rely on threshold. Some big initializers may not be converted. Use utils to test if the data is external.